spark-minestom/src/main/java/me/lucko/spark/minestom/MinestomSparkExtension.java

```diff
@@ -0,0 +1,144 @@
+import net.minestom.server.command.builder.arguments.ArgumentType;
+import net.minestom.server.command.builder.suggestion.SuggestionEntry;
+import net.minestom.server.extensions.Extension;
+import net.minestom.server.timer.ExecutionType;
+
+import java.nio.file.Path;
+import java.util.logging.Level;
+import java.util.stream.Stream;
+
+public class MinestomSparkExtension extends Extension implements SparkPlugin {
+    private SparkPlatform platform;
+    private MinestomSparkCommand command;
+
+    @Override
+    public String getVersion() {
+        return getOrigin().getVersion();
+    }
+
+    @Override
+    public Path getPluginDirectory() {
+        return getDataDirectory();
+    }
+
+    @Override
+    public String getCommandName() {
+        return "sparkms";
+    }
+
+    @Override
+    public Stream<? extends CommandSender> getCommandSenders() {
+        return Stream.concat(
+                MinecraftServer.getConnectionManager().getOnlinePlayers().stream(),
+                Stream.of(MinecraftServer.getCommandManager().getConsoleSender())
+        ).map(MinestomCommandSender::new);
+    }
+
+    @Override
+    public void executeAsync(Runnable task) {
+        MinecraftServer.getSchedulerManager().scheduleNextTick(task, ExecutionType.ASYNC);
+    }
+
+    @Override
+    public void log(Level level, String msg) {
+        if (level == Level.INFO) {
+            this.getLogger().info(msg);
+        } else if (level == Level.WARNING) {
+            this.getLogger().warn(msg);
+        } else if (level == Level.SEVERE) {
+            this.getLogger().error(msg);
+        } else {
+            throw new IllegalArgumentException(level.getName());
+        }
+    }
+
+    @Override
+    public PlatformInfo getPlatformInfo() {
+        return new MinestomPlatformInfo();
+    }
+
+    @Override
+    public ClassSourceLookup createClassSourceLookup() {
+        return new MinestomClassSourceLookup();
+    }
+
+    @Override
+    public PlayerPingProvider createPlayerPingProvider() {
+        return new MinestomPlayerPingProvider();
+    }
+
+    @Override
+    public TickReporter createTickReporter() {
+        return new MinestomTickReporter();
+    }
+
+    @Override
+    public void initialize() {
+        this.platform = new SparkPlatform(this);
+        this.platform.enable();
+        this.command = new MinestomSparkCommand(this);
+        MinecraftServer.getCommandManager().register(command);
+    }
+
+    @Override
+    public void terminate() {
+        this.platform.disable();
+        MinecraftServer.getCommandManager().unregister(command);
+    }
+
+    private static final class MinestomSparkCommand extends Command {
+        public MinestomSparkCommand(MinestomSparkExtension extension) {
+            super("sparkms");
+            setDefaultExecutor((sender, context) -> extension.platform.executeCommand(new MinestomCommandSender(sender), new String[0]));
+            var arrayArgument = ArgumentType.StringArray("query");
```

Comment:
Just a codestyle thing, please could you replace the `var`s in this method with actual types. Makes it much easier to read the code without an IDE.